.github/workflows/build.yml

```diff
@@ -317,8 +311,7 @@ jobs:
       - test-docker-k8s
     steps:
       - run: |
-          echo "Some of the required build and test workflows have failed!"
-          exit 1
+          tr -d '\n' <<< '${{ toJSON(needs.*.result) }}' | grep -q -v -E '(failure|cancelled)'
```

Comment:
Can you just pipe this to `xargs` to strip the trailing newlines?

Reply:
are you thinking 
```sh
echo '${{ toJSON(needs.*.result) }}' | xargs | grep -q -v -E '(failure|cancelled)'
```
? We could do that. I had copied this line from ci.yml, but i can change it

Reply:
Ah, okay, no need. This works just as well and there are other instances of it, so might as well keep it.